app/models/test_track/offline_session.rb

```diff
@@ -1,14 +1,4 @@
 class TestTrack::OfflineSession
-  AnalyticsAssignment = Struct.new(:visitor_id, :split_name, :variant, :context, :unsynced) do
-    def unsynced?
-      unsynced
-    end
-
-    def feature_gate?
-      split_name.end_with?('_enabled')
-    end
-  end
```

Comment:
so, remind me again why we needed this class? i swear we wouldn't have added it if it weren't necessary. we did some more refactoring so that we don't need it anymore?

Reply:
It was to get `TestTrack::Remote::Assignment` to look like `Assignment` when used by the analytics code. But there was another case that we missed... when unsynced assignments come back from the server in an online session.

We could wrap those instances in this AnalyticsAssignment wrapper, but I opted to change the contract of `TT::AnalyticsEvent` and move some of this functionality into `TT::Remote::Assignment` instead.

It feels like a win to have 2 Assignment-like classes rather than 3, but, tbh, I'm not sure if this was going somewhere else that I don't see.